language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/ExtractToConfigurableCodeAction.java

```diff
@@ -0,0 +1,345 @@
+            FunctionTypeSymbol functionTypeSymbol  = null;
+            SeparatedNodeList<?> separatedNodeList = null;
+            switch (node.kind()) {
+                case FUNCTION_CALL -> {
+                    Optional<Symbol> functionSymbol = semanticModel.symbol(node);
+                    if (functionSymbol.isEmpty()) {
+                        return Collections.emptyList();
+                    }
+                    functionTypeSymbol = ((FunctionSymbol) functionSymbol.get()).typeDescriptor();
+                    separatedNodeList = ((FunctionCallExpressionNode) node).arguments();
+                }
+                case METHOD_CALL -> {
+                    Optional<Symbol> methodCall = semanticModel.symbol(node);
+                    if (methodCall.isEmpty()) {
+                        return Collections.emptyList();
+                    }
+                    functionTypeSymbol = ((MethodSymbol) methodCall.get()).typeDescriptor();
+                    separatedNodeList = ((MethodCallExpressionNode) node).arguments();
+                }
+                case IMPLICIT_NEW_EXPRESSION -> {
+                    Optional<ParenthesizedArgList> parenthesizedArgList =
+                            ((ImplicitNewExpressionNode) node).parenthesizedArgList();
+                    if (parenthesizedArgList.isEmpty()) {
+                        return Collections.emptyList();
+                    }
+                    Optional<FunctionTypeSymbol> initType = getFunctionTypeSymbolFromNewExpr(semanticModel, node);
+                    if (initType.isEmpty()) {
+                        return Collections.emptyList();
+                    }
+                    functionTypeSymbol = initType.get();
+                    separatedNodeList = parenthesizedArgList.get().arguments();
+                }
+                case EXPLICIT_NEW_EXPRESSION -> {
+                    Optional<FunctionTypeSymbol> initType = getFunctionTypeSymbolFromNewExpr(semanticModel, node);
+                    if (initType.isEmpty()) {
+                        return Collections.emptyList();
+                    }
+                    functionTypeSymbol = initType.get();
+                    separatedNodeList = ((ExplicitNewExpressionNode) node).parenthesizedArgList().arguments();
+                }
+                default -> {
+                    return Collections.emptyList();
+                }
+            }
+
+            if (functionTypeSymbol.params().isEmpty()) {
+                return Collections.emptyList();
+            }
+            List<ParameterSymbol> parameterSymbols = functionTypeSymbol.params().get();
+            TypeSymbol anydataType = semanticModel.types().ANYDATA;
+            List<TextEdit> textEdits = new ArrayList<>();
+            boolean firstEdit = true;
+            for (int argIdx = 0; argIdx < separatedNodeList.size() && argIdx < parameterSymbols.size(); argIdx++) {
+                Node argument = separatedNodeList.get(argIdx);
```

Comment:
Don't we need to do a null check for `argument`?

Reply:
We don't. To be on the safe side I'll update.